File: 88. Merge Sorted Array/88. Merge Sorted Array.py
from typing import List


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        end1 = m - 1
        end2 = n - 1

        for i in range(m + n - 1, -1, -1):
            if end2 < 0:
                break
            elif end1 >= 0 and nums1[end1] > nums2[end2]:
                nums1[i] = nums1[end1]
                end1 -= 1
            else:
                nums1[i] = nums2[end2]
                end2 -= 1

File: 88. Merge Sorted Array/88. Merge Sorted Array.java
class Solution {

  public void merge(int[] nums1, int m, int[] nums2, int n) {
    int end1 = m - 1;
    int end2 = n - 1;

    for (int i = m + n - 1; i > -1; i--) {
      if (end2 < 0) {
        break;
      } else if (end1 >= 0 && nums1[end1] > nums2[end2]) {
        nums1[i] = nums1[end1];
        end1--;
      } else {
        nums1[i] = nums2[end2];
        end2--;
      }
    }
  }
}
